bugfix/parameterize-vocabulary-utility

shanmukhananda opened the pull request.

paul-michalik commented on CMakeLists.txt:
```diff
@@ -112,6 +112,6 @@ target_link_libraries(mono_video ${PROJECT_NAME})
-add_executable(bin_vocabulary
-tools/bin_vocabulary.cc)
-target_link_libraries(bin_vocabulary ${PROJECT_NAME})
+add_executable(vocabulary_converter
```
I understand you want to be expressive but this is too long... 😉 how about vconv or vocconv or something like that? Unix tools of that kind are usually named like icing or cconv etc...

paul-michalik replied:
It should've been "iconv" it got auto corrected by browser...

paul-michalik commented on scripts/windows/run_slam_on_garching_test_drive.bat:
```diff
@@ -5,18 +5,13 @@ set "VcpkgBinDir=%~d0\Software\vcpkg\vcpkg\installed\x86-windows-v141\bin"
-
 if not exist "%ProjectDir%\Vocabulary\ORBvoc.bin" (
-    %BinVocApp%
+    7z x "%ProjectDir%\Vocabulary\ORBvoc.bin.tar.gz" -o"%ProjectDir%\Vocabulary"
```
I haven't noticed this before, but do we assume there's always 7z installed? Do we install it ourselves?

shanmukhananda replied:
7z is installed manually. If user did not install 7z this will fail. 
Adding choco install in bootstrap.bat is one option.

paul-michalik replied:
Ah, no leave it for now. The native way would be ps ExpandArchive cmdlet, but it needs PowerShell 5 at least, so we'd have to check and install anyway.

shanmukhananda commented:
1. renamed bin_vocabulary to vocabulary_converter, txt2bin is too specific as vocabulary_converter supports xml, bin and txt conversions
2. Updated readme.md to describe vocabulary_converter
3. Uploaded ORBvoc.bin.tar.gz to repo
4. ORBvoc.bin is the default vocabulary in run scripts. 
5. using relative paths is buggy, script fails if called from different directory. fix: use absolute paths
